src/ops.rs

```diff
@@ -1213,6 +1584,31 @@ pub fn cmpxchg_native_word(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hyper
     Ok(())
 }
 
+pub fn cmpxchg8b(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Result<(), VMError>{
+        let eax = vm.get_reg(Reg32::EAX as u8, ValueSize::Dword).u32_exact()?;
+        let edx = vm.get_reg(Reg32::EDX as u8, ValueSize::Dword).u32_exact()?;
+        let ecx = vm.get_reg(Reg32::ECX as u8, ValueSize::Dword).u32_exact()?;
+        let ebx = vm.get_reg(Reg32::EBX as u8, ValueSize::Dword).u32_exact()?;
+        println!("hit in the cmpxchg opcode");
+        let address = vm.get_arg_lea(pipeline.args[0].location)?;
```

Comment:
Is there a good reason to use `get_arg_lea` and then get_mem, instead of marking the opcode as using a qword argument and using `get_arg`  and `set_arg` directly?

Reply:
the 64 bit cmpxchg doesn't use the qword directly as an argument but rather the address to the qword.